### PR TITLE
QBO Revamp: Advanced Configurations v2 API

### PR DESCRIPTION
**Pull request by Sravanksk**

### CODE REVIEW

File: apps/workspaces/apis/advanced_configurations/serializers.py [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+            'name': instance.bill_payment_account_name
+        }
+
+
+class WorkspaceScheduleSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = WorkspaceSchedule
+        fields = [
+            'enabled',
+            'start_datetime',
```
Comment (ashwin1111) [inline apps/workspaces/apis/advanced_configurations/serializers.py:59]: client will not be sending `start_datetime`. Can safely assume current time as start date time. Also, client doesn't need this info for get call.

File: apps/workspaces/apis/advanced_configurations/triggers.py [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+        )
+
+        schedule_reimbursements_sync(
+            sync_qbo_to_fyle_payments=workspace_general_settings_instance.sync_qbo_to_fyle_payments,
+            workspace_id=workspace_general_settings_instance.workspace.id
+        )
+
+    @staticmethod
+    def run_general_mappings_triggers(workspace_general_settings_instance: WorkspaceGeneralSettings):
+        schedule_bill_payment_creation(
```
Comment (ashwin1111) [inline apps/workspaces/apis/advanced_configurations/triggers.py:33]: Why is this needed? `run_workspace_general_settings_triggers` actually takes care of it

File: apps/workspaces/apis/advanced_configurations/triggers.py [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+            sync_fyle_to_qbo_payments=workspace_general_settings_instance.sync_fyle_to_qbo_payments,
+            workspace_id=workspace_general_settings_instance.workspace.id
+        )
+
+    @staticmethod
+    def run_workspace_schedule_triggers(workspace_schedule: WorkspaceSchedule):
+        schedule_sync(
```
Comment (ashwin1111) [inline apps/workspaces/apis/advanced_configurations/triggers.py:40]: We are doing same operations twice

**Review by ashwin1111 — APPROVED** [review]

File: apps/workspaces/apis/advanced_configurations/views.py [hunk 4]
```diff
@@ -0,0 +1,13 @@
+from rest_framework import generics
+
+from apps.workspaces.models import Workspace
+
+from .serializers import AdvancedConfigurationsSerializer
+
+
+class AdvancedConfigurationsView(generics.RetrieveUpdateAPIView):
+    serializer_class = AdvancedConfigurationsSerializer
+    lookup_field = 'workspace_id'
```
Comment (Shwetabhk) [inline apps/workspaces/apis/advanced_configurations/views.py:10]: lookup field is not required I think

File: apps/workspaces/apis/advanced_configurations/serializers.py [hunk 5]
```diff
@@ -0,0 +1,132 @@
...
+
+        GeneralMapping.objects.update_or_create(
+            workspace=instance,
+            defaults={
+                'bill_payment_account_name': general_mappings.get('bill_payment_account').get('name'),
+                'bill_payment_account_id': general_mappings.get('bill_payment_account').get('id')
+            }
+        )
+
+        workspace_schedule_instance, _ = WorkspaceSchedule.objects.update_or_create(
```
Comment (Shwetabhk) [inline apps/workspaces/apis/advanced_configurations/serializers.py:109]: call `schedule_sync` here
trigger is not required

File: apps/workspaces/apis/advanced_configurations/serializers.py [hunk 6]
```diff
@@ -0,0 +1,132 @@
...
+            workspace=instance,
+            defaults={
+                'enabled': workspace_schedules.get('enabled'),
+                'interval_hours': workspace_schedules.get('interval_hours')
+            }
+        )
+
+        AdvancedConfigurationsTriggers.run_workspace_general_settings_triggers(workspace_general_settings_instance)
+        AdvancedConfigurationsTriggers.run_workspace_schedule_triggers(workspace_schedule_instance)
```
Comment (Shwetabhk) [inline apps/workspaces/apis/advanced_configurations/serializers.py:118]: remove this trigger

File: apps/workspaces/apis/advanced_configurations/serializers.py [hunk 7]
```diff
@@ -0,0 +1,130 @@
...
+        GeneralMapping.objects.update_or_create(
+            workspace=instance,
+            defaults={
+                'bill_payment_account_name': general_mappings.get('bill_payment_account').get('name'),
+                'bill_payment_account_id': general_mappings.get('bill_payment_account').get('id')
+            }
+        )
+
+        schedule_sync(
+            workspace_id=workspace_general_settings_instance.workspace.id,
```
Comment (Shwetabhk) [inline apps/workspaces/apis/advanced_configurations/serializers.py:111]: can directrly do `instance.id`

**Review by Shwetabhk — APPROVED** [review]
